src/shared/utilities/promptSetting.ts

```diff
@@ -0,0 +1,37 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { getLogger } from '../logger'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+export type PROMPT_NAME =
+    | 'warnBeforeRunCommand'
+    | 'warnBeforeEnablingExcecuteCommand'
+    | 'warnBeforeDisablingExecuteCommand'
+export const DEFAULT_PROMPTS: PromptSetting = {
+    warnBeforeRunCommand: true,
+    warnBeforeEnablingExcecuteCommand: true,
+    warnBeforeDisablingExecuteCommand: true,
+}
+
+export interface PromptSetting {
+    [promptName: string]: boolean
```

Comment:
This interface is a narrowing of the TS object type since it restricts any property to be a boolean, so it isn't entirely pointless. This interface can potentially be much more useful when TS 4.4 is released, allowing the index signature to be further narrowed with the constant prompt name alias.

This type does not need to be exported though since it just serves to keep things internally consistent. Same goes for `PROMPT_NAME` unless there is a separate module that is manipulating arbitrary prompt names? Callers of `disablePrompt` do not need to know how the data is being represented within the settings.

Reply:
After looking over the code again I see that the type is only being used to read from settings, so yes I would say that just assuming the settings has the correct shape is poor practice.